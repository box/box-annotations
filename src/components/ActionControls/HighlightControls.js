// @flow
import React from 'react';
import classNames from 'classnames';

import PlainButton from 'box-ui-elements/es/components/plain-button';
import IconHighlightAnnotation from 'box-ui-elements/es/icons/annotations/IconHighlightAnnotation';
import IconHighlightCommentAnnotation from 'box-ui-elements/es/icons/annotations/IconHighlightCommentAnnotation';

import './HighlightControls.scss';

const CLASS_HIGHLIGHT_CONTROLS = 'ba-action-controls-highlight';
const CLASS_HIGHLIGHT_BTN = 'ba-highlight-btn';
const CLASS_HIGHLIGHT_COMMENT_BTN = 'ba-highlight-comment-btn';
const CLASS_SAVED_HIGHLIGHT = 'ba-saved-highlight';

type Props = {
    canAnnotateAndDelete: boolean,
    canComment: boolean,
    isPending?: boolean,
    onCommentClick: Function,
    onCreate: Function,
    onDelete: Function,
};

const HighlightControls = ({
    canAnnotateAndDelete,
    canComment,
    onCreate,
    onDelete,
    onCommentClick,
    isPending,
}: Props) => (
    <div className={CLASS_HIGHLIGHT_CONTROLS}>
        {canAnnotateAndDelete && (
            <PlainButton className={CLASS_HIGHLIGHT_BTN} onClick={isPending ? onCreate : onDelete} type="button">
                <IconHighlightAnnotation
                    className={classNames({
                        [CLASS_SAVED_HIGHLIGHT]: !isPending,
                    })}
                    height={20}
                    width={20}
                />
            </PlainButton>
        )}
        {canComment && (
            <PlainButton className={CLASS_HIGHLIGHT_COMMENT_BTN} onClick={onCommentClick} type="button">
                <IconHighlightCommentAnnotation height={20} width={20} />
            </PlainButton>
        )}
    </div>
);

HighlightControls.defaultProps = {
    canAnnotateAndDelete: false,
    isPending: false,
};

export default HighlightControls;
