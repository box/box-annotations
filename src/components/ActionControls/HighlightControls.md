### Description
Component to represent the highlight action buttons in the annotation dialog

### Examples
```jsx
const HighlightControlsExamples = require('examples').HighlightControlsExamples;

<HighlightControlsExamples />
```
