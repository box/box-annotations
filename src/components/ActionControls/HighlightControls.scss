@import '../../variables';

.ba {
    .ba-action-controls-highlight {
        $button-spacing: 5px;

        display: flex;
        margin-left: -#{$button-spacing};
        margin-right: -#{$button-spacing};

        .btn-plain {
            flex-grow: 1;
            padding-left: $button-spacing;
            padding-right: $button-spacing;

            .icon-annotation-highlight,
            .icon-annotation-highlight-comment .icon {
                fill: $sevens;
            }

            .icon-annotation-highlight.ba-saved-highlight {
                fill: $highlight-yellow;
            }
        }

        .btn-plain:hover {
            .icon-annotation-highlight,
            .icon-annotation-highlight-comment .icon {
                fill: $better-black;
            }

            .icon-annotation-highlight.ba-saved-highlight {
                fill: $highlight-yellow-hover;
            }
        }
    }

    // MOBILE & TABLET CSS
    @media #{$mobile}, #{$tablet} {
        .ba-inline-popover .ba-action-controls-highlight {
            background: white;
            border-top: 1px solid $seesee;
            bottom: 0;
            box-shadow: 0 2px 6px 0 rgba(0, 0, 0, .1);
            left: 0;
            margin: 0;
            padding: 15px;
            position: fixed;
            width: 100%;
        }

        .ba-action-controls-highlight .btn-plain,
        .ba-action-controls-highlight .btn-plain:hover {
            padding: 10px;
        }

        .ba-highlight-comment-btn,
        .ba-highlight-comment-btn:hover {
            border-left: 1px solid $seesee;
        }
    }
}
