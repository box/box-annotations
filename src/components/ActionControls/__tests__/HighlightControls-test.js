import * as React from 'react';
import { shallow } from 'enzyme';

import HighlightControls from '../HighlightControls';

const onCreate = jest.fn();
const onCommentClick = jest.fn();

describe('components/ActionControls/HighlightControls', () => {
    const render = (props = {}) =>
        shallow(<HighlightControls onCreate={onCreate} onCommentClick={onCommentClick} {...props} />);

    test('should correctly render pending plain highlight controls', () => {
        const wrapper = render({
            canAnnotateAndDelete: true,
            canComment: false,
            isPending: true
        });
        expect(wrapper).toMatchSnapshot();
    });

    test('should correctly render pending highlight comment controls', () => {
        const wrapper = render({
            canAnnotateAndDelete: true,
            canComment: true,
            isPending: true
        });
        expect(wrapper).toMatchSnapshot();
    });

    test('should correctly render saved plain highlight controls', () => {
        const wrapper = render({
            canAnnotateAndDelete: true,
            canComment: false
        });
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.ba-saved-highlight').length).toEqual(1);
    });

    test('should correctly render saved highlight comment controls', () => {
        const wrapper = render({
            canAnnotateAndDelete: true,
            canComment: true
        });
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.ba-saved-highlight').length).toEqual(1);
    });
});
