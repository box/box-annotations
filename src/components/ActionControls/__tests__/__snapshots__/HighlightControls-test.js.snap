// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ActionControls/HighlightControls should correctly render pending highlight comment controls 1`] = `
<div
  className="ba-action-controls-highlight"
>
  <PlainButton
    className="ba-highlight-btn"
    onClick={[MockFunction]}
    type="button"
  >
    <IconHighlightAnnotation
      className=""
      height={20}
      width={20}
    />
  </PlainButton>
  <PlainButton
    className="ba-highlight-comment-btn"
    onClick={[MockFunction]}
    type="button"
  >
    <IconHighlightCommentAnnotation
      className=""
      height={20}
      width={20}
    />
  </PlainButton>
</div>
`;

exports[`components/ActionControls/HighlightControls should correctly render pending plain highlight controls 1`] = `
<div
  className="ba-action-controls-highlight"
>
  <PlainButton
    className="ba-highlight-btn"
    onClick={[MockFunction]}
    type="button"
  >
    <IconHighlightAnnotation
      className=""
      height={20}
      width={20}
    />
  </PlainButton>
</div>
`;

exports[`components/ActionControls/HighlightControls should correctly render saved highlight comment controls 1`] = `
<div
  className="ba-action-controls-highlight"
>
  <PlainButton
    className="ba-highlight-btn"
    type="button"
  >
    <IconHighlightAnnotation
      className="ba-saved-highlight"
      height={20}
      width={20}
    />
  </PlainButton>
  <PlainButton
    className="ba-highlight-comment-btn"
    onClick={[MockFunction]}
    type="button"
  >
    <IconHighlightCommentAnnotation
      className=""
      height={20}
      width={20}
    />
  </PlainButton>
</div>
`;

exports[`components/ActionControls/HighlightControls should correctly render saved plain highlight controls 1`] = `
<div
  className="ba-action-controls-highlight"
>
  <PlainButton
    className="ba-highlight-btn"
    type="button"
  >
    <IconHighlightAnnotation
      className="ba-saved-highlight"
      height={20}
      width={20}
    />
  </PlainButton>
</div>
`;
